tests/test_file_perm.py

```diff
@@ -0,0 +1,25 @@
+import unittest
+import stat
+
+from mock import Mock, patch
+
+from patroni.file_perm import pg_perm
+
+
+class TestFilePermissions(unittest.TestCase):
+
+    @patch('os.stat')
+    @patch('os.umask')
+    @patch('patroni.file_perm.logger.error')
+    def test_set_umask(self, mock_logger, mock_umask, mock_stat):
+        mock_umask.side_effect = Exception
+        mock_stat.return_value.st_mode = stat.S_IRWXU | stat.S_IRGRP | stat.S_IXGRP
+        pg_perm.set_permissions_from_data_directory('test')
+        self.assertEqual(mock_umask.call_args[0][0], stat.S_IWGRP | stat.S_IRWXO)
+        self.assertEqual(mock_logger.call_args[0][0], 'Can not set umask to %03o: %r')
+
+    @patch('os.stat', Mock(side_effect=FileNotFoundError))
+    @patch('patroni.file_perm.logger.error')
+    def test_set_permissions_from_data_directory(self, mock_logger):
+        pg_perm.set_permissions_from_data_directory('test')
+        self.assertEqual(mock_logger.call_args[0][0], 'Can not check permissions on %s: %r')
```

Comment:
What about adding a couple tests to check if `__set_group_permissions` or `__set_owner_permissions` is being called?

Reply:
Well, `umask` could only called when `os.stat()` didn't raise any exception.
Therefore one or another is successfully called :)

Reply:
Ah, sure. I guess I didn't express myself clearly, sorry.
I meant based on a few sample mock values for `st_mode`, to check if it's taking the expected code path.

Reply:
At least one path is checked here: https://github.com/zalando/patroni/pull/2781/files#diff-6251748fa4c45a038329a6f9eb925a57f7a1c07b50b64823ca36ea20804748efR16-R18

Reply:
Right, the group one.
Your call on adding a test for the other or not.